pkg/scheduler/cache/cache.go

```diff
@@ -240,8 +242,112 @@ func (cache *schedulerCache) addPod(pod *v1.Pod) {
 	n.AddPod(pod)
 }
 
+// this function expects valid pod, and valid, non-empty resizeRequestAnnotation json string
+func (cache *schedulerCache) getPodResizeRequirements(pod *v1.Pod, resizeRequestAnnotation string) (map[string]v1.Container, *Resource, error) {
```

Comment:
this could use a unit test

Reply:
yes i plan to add this after the merge with subsequent bugfixes. this is currently covered indirectly by the newly added unit tests.